src/buildReader.ts

```diff
@@ -323,9 +324,18 @@ export default function buildReader(types: readonly RosMsgDefinition[]): Seriali
         // This fully deserializes all fields of the message into native types
         // Typed arrays are considered native types and remain as typed arrays
         toJSON() {
-          return {
-            ${fields.join(",\n")}
-          };
+          const view = this.#view;
+          const buffer = new DataView(view.buffer, view.byteOffset, view.byteLength);
+          return fullReader.readMessage(buffer)
```

Comment:
nit: semicolons on these returns

Reply:
What's the point of making a new DataView given all the same arguments? can we just `readMessage(this.#view)`?

Reply:
oh fascinating - the full reader uses Uint8Array - but I guess it just grabs the buffer out of that so passing in DataView works too - this is technically a typo and I meant to make a new Uint8Array here to align with the readMessage api from MessageReader

Reply:
In my mcap work I did notice some weird behavior with TypeScript treating some of these classes as compatible when they really aren’t 😕 For example I think I was allowed to pass a Uint8Array to the DataView constructor when it really wants an ArrayBuffer.